### PR TITLE
[Refactor] Update Artist/Consign to support MP-based data

### PR DESCRIPTION
**Pull request by damassi**
Depends on https://github.com/artsy/metaphysics/pull/2266

In the MP pr above the target supply logic was moved over to MP so that it could be shared across platforms, as well as for scale as we extend our microfunnel "top 20" artists to the full target supply list. 

In this PR we update all the data paths, and ensure that our graphql data is co-located where its called, versus passing throughout the tree. 

Was also able to get in the recently sold artworks randomization in as that logic has moved up to the data layer, thus bypassing SSR double render considerations. 

Once this PR is merged will move over the Eigen and wire up the button there, and then move back over to MP to start thinking about getting raw RedShift data in. 

#### Screenshots of refactor working

Screen Shot 2020-03-31 at 12 22 16 AM
Screen Shot 2020-03-31 at 12 22 28 AM

(Side note: `git commit` often! I lost the entirety of this work one time around through some kind of FS freakout that started deleting all of the files in reaction... since nothing was committed, the code was all gone.....no clue.)

### CODE REVIEW

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/index.tsx [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { Media } from "Utils/Responsive"
+import { ArtistConsignHeaderImagesFragmentContainer as ArtistConsignHeaderImages } from "./ArtistConsignHeaderImages"
 
 interface ArtistConsignHeaderProps {
-  artistConsignment: ArtistConsignment
-  artistName: string
-  artworksByInternalID: Consign_artworksByInternalID
+  artist: ArtistConsignHeader_artist
```
Comment (pepopowitz) [inline src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/index.tsx]: 👍

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignMeta.tsx [hunk 2]
```diff
@@ -1,33 +1,67 @@
...
+    artist: {
+      name,
+      href,
+      targetSupply: {
+        microfunnel: { artworks },
+      },
+    },
+  } = props
+
+  const imageURL = artworks[0].artwork.image.imageURL
```
Comment (pepopowitz) [inline src/Apps/Artist/Routes/Consign/Components/ArtistConsignMeta.tsx]: Another place for optional chaining?

File: src/Apps/Artist/Routes/Consign/index.tsx [hunk 3]
```diff
@@ -1,67 +1,32 @@
...
-      />
-      <ArtistConsignPageViews
-        artistConsignment={artistConsignment}
-        artistName={artist.name}
-      />
-      <ArtistConsignMarketTrends
-        artistConsignment={artistConsignment}
-        artistID={match.params.artistID}
-      />
+      <ArtistConsignMeta artist={artist} />
```
Comment (pepopowitz) [inline src/Apps/Artist/Routes/Consign/index.tsx]: 👌

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
 interface HeaderImageProps {
-  artworksByInternalID: Consign_artworksByInternalID
+  artist: ArtistConsignHeaderImages_artist
 }
 
-export const HeaderImages: React.FC<HeaderImageProps> = props => {
+export const ArtistConsignHeaderImages: React.FC<HeaderImageProps> = props => {
   const { error, leftImage, rightImage } = getRandomImages(
-    props.artworksByInternalID
+    props.artist.targetSupply.microfunnel.artworks
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx]: If a user isn't a part of the microfunnel target supply this route will never execute. Its guarded at the top-most [route level here](https://github.com/artsy/reaction/blob/8f4cdd3b3e994629ab09559786c9b9318aff67d1/src/Apps/Artist/routes.tsx#L251-L255).

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx [hunk 5]
```diff
@@ -32,37 +35,60 @@ export const HeaderImages: React.FC<HeaderImageProps> = props => {
...
   try {
     if (loop > 100) {
       throw new Error("No suitable images found, exiting")
     }
 
     // Layout requires that we only use vertically oriented images
-    const widthExceedsHeight = ({ image }) => {
-      return image.resized.width > image.resized.height
+    const widthExceedsHeight = ({ artwork }) => {
+      return artwork.image.resized.width > artwork.image.resized.height
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx]: This function lives inside of a try catch. If there's an issue, we throw and return an error, as one often sees in Go.

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/__tests__/ArtistConsignHeaderImages.test.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            },
+          },
+        },
+      ]
+
+      const { error } = getRandomImages(artworks)
+      expect(error).toEqual(true)
+    })
+
+    it("errors out if no images found", () => {
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/__tests__/ArtistConsignHeaderImages.test.tsx]: The design is `images` or `error`, but will always have images, unless there's an ephemeral problem upstream in which this non-breaking error will guard, signaling a `null` return at the component level.

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignMeta.tsx (hunk 2 above)
Comment (damassi) [reply src/Apps/Artist/Routes/Consign/Components/ArtistConsignMeta.tsx]: Any artist in "top 20" will have an image, unless there's a problem at the service level. We now have a proper [ErrorBoundary](https://github.com/artsy/reaction/blob/master/src/Components/ErrorBoundary.tsx) up the stack which will handle cases like this.

File: src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx [hunk 7]
```diff
@@ -32,37 +35,60 @@ export const HeaderImages: React.FC<HeaderImageProps> = props => {
...
 /**
  * Iterate over "Recently Purchased" artworks and randomly select two. If there
  * are no images found, or the dimension requirements are not suitable, return
  * an error indicator and bail.
  */
-export const getRandomImages = (
-  artworksByInternalID: Consign_artworksByInternalID,
-  loop: number = 0
-) => {
+export const getRandomImages = (artworks: Artworks, loop: number = 0) => {
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader/ArtistConsignHeaderImages.tsx]: It seems like reacts algorithm is intelligent enough here and things seem fine.

File: src/Apps/Artist/Routes/Consign/__tests__/ConsignRoute.test.tsx [hunk 8]
```diff
@@ -91,7 +59,7 @@ describe("ConsignRoute", () => {
   })
 
   describe("ArtistConsignHeader", () => {
-    it("displays artist name in header", async () => {
+    it.only("displays artist name in header", async () => {
```
Comment (eessex) [inline src/Apps/Artist/Routes/Consign/__tests__/ConsignRoute.test.tsx]: `only` intentional?

Comment (damassi) [reply src/Apps/Artist/Routes/Consign/__tests__/ConsignRoute.test.tsx]: Ah, nice catch 👍

**Comment by artsyit** [comment]
:rocket: PR was released in `v25.41.0` :rocket:

**Pull request merged**